plutus-ir/test/ParserSpec.hs

```diff
@@ -0,0 +1,119 @@
+{-# LANGUAGE FlexibleContexts #-}
+module ParserSpec (parsing) where
+
+import           PlutusPrelude
+
+import           Common
+import           TestLib
+
+import           Control.Monad
+import           Control.Monad.Except
+import           Control.Monad.State
+
+import qualified Data.ByteString.Lazy         as BSL
+
+import           Language.PlutusIR
+import qualified Language.PlutusIR.Parser     as Parser
+
+import qualified Language.PlutusCore          as PLC
+import qualified Language.PlutusCore.Constant as PLC
+import qualified Language.PlutusCore.Pretty   as PLC
+
+import           Hedgehog                     hiding (Var)
+import qualified Hedgehog.Gen                 as Gen
+import qualified Hedgehog.Range               as Range
+
+import           Text.Megaparsec              hiding (failure)
+
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
+
+propParser :: Property
+propParser = property $ do
+    prog <- forAll genProgram
+    let pretty = prettyString prog
+    ast <- case Parser.parse "test" Parser.program pretty of
+               Left e -> do
+                   annotate pretty
+                   annotate $ parseErrorPretty e
+                   failure
+               Right p -> pure p
+    pretty === prettyString ast
+
+parsing :: TestNested
+parsing = return $ testGroup "parsing"
+    [testProperty "parser round-trip" propParser]
+
+
+genTyName :: MonadGen m => m (TyName ())
+genTyName = TyName <$> genName
+
+-- TODO make this robust against generating identfiers such as "fix"?
```

Comment:
Yes, this is somewhat important, see the recent https://github.com/input-output-hk/plutus/pull/479